File: Aula05_06-09-23/selection-sort.py
from nomes_desord_reduzido import nomes
global pass_, comps, swap

def selection_sort(vetor, fn_comp): 
    pass_, comps, swap = 0, 0, 0 
    for selected in range(len(vetor) - 1):
        pass_ += 1
        menor = selected + 1
        for i in range((menor + 1), len(vetor)):
            if fn_comp(vetor[menor], vetor[i]):
                menor = i
                comps += 1
        if fn_comp(vetor[selected], vetor[menor]):
            vetor[selected], vetor[menor] = vetor[menor], vetor[selected]
            swap += 1
    print(pass_, comps, swap)
    return vetor



print(selection_sort(vetor=nomes, fn_comp=lambda obj1, obj2: obj1 > obj2))

# def fn_comp(obj1, obj2):
#     return obj1 > obj2


# lista_nomes = nomes
# print(selection_sort(lista_nomes, fn_comp))

File: Aula05_06-09-23/selection-sort.mjs
let pass, comps, troca

function selectionSort(vetor){
    pass=0, comps=0, troca=0
    for(let selecionada = 0; selecionada < vetor.length -1; selecionada++){
        pass++
        let menor = selecionada + 1
        for(let i = menor + 1; i < vetor.length; i++){
            if(vetor[menor] > vetor[i]){
                menor = i
                comps++
            }
        }
        if(vetor[selecionada] > vetor[menor]){
            [vetor[selecionada], vetor[menor]] = [vetor[menor], vetor[selecionada]]
            troca++
        }
    }
}

let nums = [77, 44, 55, 33, 22, 11, 10, 3, 1]
selectionSort(nums)
console.log(nums)
console.log(pass, comps, troca)
